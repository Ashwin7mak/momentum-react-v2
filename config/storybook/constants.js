module.exports = {
  REPO_ROOT: '../../../',
  SRC_DIR: './src/',
  ESM_DIR: './dist/esm/',
  CJS_DIR: './dist/cjs/',
  COMPONENTS_DIR: './components/',
};
