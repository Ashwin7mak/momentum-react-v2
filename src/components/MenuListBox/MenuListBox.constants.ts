const CLASS_PREFIX = 'md-menu-list-box';

const STYLE = {
  wrapper: `${CLASS_PREFIX}-wrapper`,
};

export { STYLE };
