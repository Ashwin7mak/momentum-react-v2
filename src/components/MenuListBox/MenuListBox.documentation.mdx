`<MenuListBox />` component is a Menu Background for Select and Menu List.
