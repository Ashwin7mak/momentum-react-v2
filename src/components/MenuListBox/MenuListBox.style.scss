.md-menu-list-box-wrapper {
  min-width: 16rem;
  position: absolute;
  margin: 0.25rem 0 0 0;
  padding: 0;
  list-style: none;
  background: var(--theme-background-solid-primary-normal);
  padding: 0.5rem;
  border: 1px solid var(--codeinput-border);
  border-radius: 0.75rem;
  box-sizing: border-box;
  transition: background-color 0.2s, color 0.2s, border-color 0.2s;

  &:focus {
    box-shadow: var(--focus-ring-box-shadow);
    outline: var(--focus-ring-outline);
  }
}
