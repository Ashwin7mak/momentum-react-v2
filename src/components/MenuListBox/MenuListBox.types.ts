import { ReactNode } from 'react';

export interface Props {
  /**
   * className prop description
   */
  className?: string;
  /**
   * children props
   */
  children?: ReactNode;
}
