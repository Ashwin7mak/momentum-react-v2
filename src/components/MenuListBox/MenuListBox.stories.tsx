import React, { FC } from 'react';
import { Story } from '@storybook/react';

import MenuListBox, { MenuListBoxProps } from './';
import {
  Title,
  Subtitle,
  Description,
  Primary,
  ArgsTable,
  PRIMARY_STORY,
} from '@storybook/addon-docs';

import Documentation from './MenuListBox.documentation.mdx';
import Flex from '../Flex';
import argTypes from './MenuListBox.stories.args';

const DocsPage: FC = () => (
  <>
    <Title />
    <Subtitle />
    <Description />
    <Documentation />
    <Primary />
    <ArgsTable story={PRIMARY_STORY} />
  </>
);

export default {
  title: 'Momentum UI/MenuListBox',
  component: MenuListBox,
  parameters: {
    expanded: true,
    docs: {
      page: DocsPage,
    },
  },
  argTypes: {
    ...argTypes,
  },
};

const Template: Story<MenuListBoxProps> = (args) => (
  <MenuListBox {...args} style={{ width: '20rem', height: '20rem' }}>
    <Flex justifyContent="center" alignItems="center" style={{ height: '100%' }}>
      This is a Menu Background
    </Flex>
  </MenuListBox>
);

const Example = Template.bind({});

export { Example };
