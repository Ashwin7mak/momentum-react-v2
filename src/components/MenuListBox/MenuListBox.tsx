import React, { RefObject, forwardRef } from 'react';
import classnames from 'classnames';

import './MenuListBox.style.scss';
import { Props } from './MenuListBox.types';
import { STYLE } from './MenuListBox.constants';

//TODO: Implement notch version, Also probably needs renaming
//TODO: Add elevation / shadow when tokens available

const MenuListBox = (props: Props, ref: RefObject<HTMLUListElement>) => {
  const { className, children, ...rest } = props;

  return (
    <ul {...rest} ref={ref} className={classnames(className, STYLE.wrapper)}>
      {children}
    </ul>
  );
};

/**
 * Menu Background for Select and Menu List
 */

const _MenuListBox = forwardRef(MenuListBox);
_MenuListBox.displayName = 'MenuListBox';

export default _MenuListBox;
