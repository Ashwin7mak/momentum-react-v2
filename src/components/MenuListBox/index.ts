import { default as MenuListBox } from './MenuListBox';
import { Props } from './MenuListBox.types';
import * as CONSTANTS from './MenuListBox.constants';

export { CONSTANTS as MENU_LIST_BOX_CONSTANTS };

export type MenuListBoxProps = Props;

export default MenuListBox;
