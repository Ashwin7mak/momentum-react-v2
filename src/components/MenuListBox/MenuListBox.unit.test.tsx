import MenuListBox from '.';
import { mount } from 'enzyme';
import React from 'react';
import { STYLE } from './MenuListBox.constants';

describe('MenuListBox', () => {
  let container;

  describe('snapshot', () => {
    it('should match snapshot', () => {
      expect.assertions(1);

      container = mount(<MenuListBox>Test</MenuListBox>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with className', () => {
      expect.assertions(1);

      const className = 'example-class';

      container = mount(<MenuListBox className={className}>Test</MenuListBox>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with id', () => {
      expect.assertions(1);

      const id = 'example-id';

      container = mount(<MenuListBox id={id}>Test</MenuListBox>);

      expect(container).toMatchSnapshot();
    });

    it('should match snapshot with style', () => {
      expect.assertions(1);

      const style = { color: 'pink' };

      container = mount(<MenuListBox style={style}>Test</MenuListBox>);

      expect(container).toMatchSnapshot();
    });
  });

  describe('attributes', () => {
    it('should have its wrapper class', async () => {
      container = mount(<MenuListBox>Test</MenuListBox>);

      const element = container.find(MenuListBox).getDOMNode();

      expect(element.classList.contains(STYLE.wrapper));
    });

    it('should have provided class when className is provided', () => {
      expect.assertions(1);

      const className = 'example-class';

      container = mount(<MenuListBox className={className}>Test</MenuListBox>);

      const element = container.find(MenuListBox).getDOMNode();

      expect(element.classList.contains(className)).toBe(true);
    });

    it('should have provided id when id is provided', () => {
      expect.assertions(1);

      const id = 'example-id';

      container = mount(<MenuListBox id={id}>Test</MenuListBox>);

      const element = container.find(MenuListBox).getDOMNode();

      expect(element.id).toBe(id);
    });

    it('should have provided style when style is provided', () => {
      expect.assertions(1);

      const style = { color: 'pink' };
      const styleString = 'color: pink;';

      container = mount(<MenuListBox style={style}>Test</MenuListBox>);

      const element = container.find(MenuListBox).getDOMNode();

      expect(element.getAttribute('style')).toBe(styleString);
    });
  });
});
