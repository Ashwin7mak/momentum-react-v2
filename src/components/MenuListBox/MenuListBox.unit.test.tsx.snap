// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuListBox snapshot should match snapshot 1`] = `
<MenuListBox>
  <ul
    className="md-menu-list-box-wrapper"
  >
    Test
  </ul>
</MenuListBox>
`;

exports[`MenuListBox snapshot should match snapshot with className 1`] = `
<MenuListBox
  className="example-class"
>
  <ul
    className="example-class md-menu-list-box-wrapper"
  >
    Test
  </ul>
</MenuListBox>
`;

exports[`MenuListBox snapshot should match snapshot with id 1`] = `
<MenuListBox
  id="example-id"
>
  <ul
    className="md-menu-list-box-wrapper"
    id="example-id"
  >
    Test
  </ul>
</MenuListBox>
`;

exports[`MenuListBox snapshot should match snapshot with style 1`] = `
<MenuListBox
  style={
    Object {
      "color": "pink",
    }
  }
>
  <ul
    className="md-menu-list-box-wrapper"
    style={
      Object {
        "color": "pink",
      }
    }
  >
    Test
  </ul>
</MenuListBox>
`;
